DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/RootBaseObjectBuilder.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rhino.DocObjects;
+using Rhino;
+using Speckle.Connectors.DUI.Bindings;
+using Speckle.Core.Models;
+using Speckle.Core.Kits;
+using System.Threading;
+
+namespace Speckle.Connectors.Rhino7.Operations.Send;
+
+/// <summary>
+/// Stateless builder object to turn an <see cref="ISendFilter"/> into a <see cref="Base"/> object
+/// </summary>
+public class RootBaseObjectBuilder
+{
+  private readonly Func<ISpeckleConverter> _converterFactory;
+
+  public RootBaseObjectBuilder(Func<ISpeckleConverter> converterFactory)
+  {
+    _converterFactory = converterFactory;
+  }
+
+  public Base Build(
+    ISendFilter sendFilter,
+    Action<string, double?>? onOperationProgressed = null,
+    CancellationToken ct = default
+  )
+  {
+    List<RhinoObject> rhinoObjects = sendFilter
+      .GetObjectIds()
+      .Select(id => RhinoDoc.ActiveDoc.Objects.FindId(new Guid(id)))
+      .Where(obj => obj != null)
+      .ToList();
+
+    if (rhinoObjects.Count == 0)
+    {
+      throw new InvalidOperationException("No objects were found. Please update your send filter!");
+    }
+
+    ISpeckleConverter converter = _converterFactory();
+
+    converter.SetContextDocument(RhinoDoc.ActiveDoc);
+    Base commitObject = ConvertObjects(rhinoObjects, converter, onOperationProgressed, ct);
+
+    return commitObject;
+  }
+
+  private static Collection ConvertObjects(
+    List<RhinoObject> rhinoObjects,
+    ISpeckleConverter converter,
+    Action<string, double?>? onOperationProgressed = null,
+    CancellationToken ct = default
+  )
+  {
+    var rootObjectCollection = new Collection { name = RhinoDoc.ActiveDoc.Name ?? "Unnamed document" };
+    int count = 0;
+
+    Dictionary<int, Collection> layerCollectionCache = new();
+    // TODO: Handle blocks.
```

Comment:
Let's get out of the habit of adding more TODOs. Will also suggest to the team but let's use POC: instead as it's easy to distinguish existing to new